README.md

```diff
@@ -515,10 +528,10 @@ conan build ..
 ```bash
 # Inside <path to onemkl>
 mkdir build && cd build
-export CXX=<path_to_dpcpp_compiler>/bin/dpcpp;
```

Comment:
In line 78 of new changes, we have `if(CMAKE_CXX_COMPILER)`. Since `CMAKE_CXX_COMPILER` variable is cached by CMake, their documentation recommends using CC, FC and CXX to define compilers in the environment instead of defining `CMAKE_CXX_COMPILER` manually (it is still a supported mechanism). Can we keep the recommended `export CXX` approach and verify that this method works?

Reply:
Because now we want to auto-select compilers, we will encounter difficulties using this approach.

First, `CMAKE_CXX_COMPILER` is cached after `project` in CMakeLists.txt, but CMake also detects CXX compiler automatically after `project`. If we rely on cached `CMAKE_CXX_COMPILER`, we cannot tell if it is supplied by users or CMake.

Next, it is possible to detect `CXX` as an environment variable before `project`, but we may have a situation that the user sets `CXX` variable for other builds, then this `CXX` will be picked up mistakenly. I think it's safer to set CXX compiler in a self-contained way if necessary.

Reply:
got it, thanks!